Scripts/Runtime/Core/Util/MathExtension.cs

```diff
@@ -0,0 +1,406 @@
+using System.Runtime.CompilerServices;
+using Anvil.CSharp.Mathematics;
+using Unity.Mathematics;
+
+namespace Anvil.Unity.Core
+{
+    /// <summary>
+    /// A collection of extension methods for working with types under the <see cref="Unity.Mathematics"/> namespace.
+    /// (float3, int2, etc..)
+    /// </summary>
+    public static class MathExtension
+    {
+        /// <summary>
+        /// Get the inverse of a <see cref="float3"/>.
+        /// </summary>
+        /// <remarks>
+        /// Any components that are 0 will invert to <see cref="float.NaN"/>.
+        /// Use <see cref="GetInverseSafe" /> if this is a possibility.
+        /// </remarks>
+        /// <param name="value">The value to get the inverse of.</param>
+        /// <returns>The inverted value</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static float3 GetInverse(this float3 value)
+        {
+            return 1f / value;
+        }
+
+        /// <summary>
+        /// Get the inverse of a <see cref="float3"/> with any <see cref="float.NaN"/> or infinite components set to 0
+        /// </summary>
+        /// <param name="value">The value to get the inverse of.</param>
+        /// <returns>The inverted value</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static float3 GetInverseSafe(this float3 value)
+        {
+            float3 inverse = value.GetInverse();
+            return math.select(inverse, float3.zero, math.isinf(inverse) | math.isnan(inverse));
+        }
+
+        /// <summary>
+        /// Checks if two <see cref="float"/> values are approximately equal according to the
+        /// <see cref="MathUtil.FLOATING_POINT_EQUALITY_TOLERANCE"/>.
+        /// </summary>
+        /// <param name="a">The <see cref="float"/> to compare.</param>
+        /// <param name="b">The <see cref="float"/> to compare.</param>
+        /// <returns>
+        /// True if the delta between <see cref="a"/> and <see cref="b"/> is less than
+        /// <see cref="MathUtil.FLOATING_POINT_EQUALITY_TOLERANCE"/>.
+        /// </returns>
+        /// <remarks>
+        /// Resolves infinite and NaN comparisons as false with improved performance vs <see cref="IsApproximately"/>.
```

Comment:
I think all of these should be referencing `IsApproximatelySafe`?

Reply:
Good catch. Fixed